lib/conduit_async.ml

```diff
@@ -22,12 +22,44 @@ IFDEF HAVE_ASYNC_SSL THEN
 open Async_ssl.Std
 END
 
+module Ssl = struct
+IFDEF HAVE_ASYNC_SSL THEN
+  type config = {
+    version : Ssl.Version.t option;
+    name : string option;
+    ca_file : string option;
+    ca_path : string option;
+    session : Ssl.Session.t option sexp_opaque;
+    verify : (Ssl.Connection.t -> bool Deferred.t) option;
+  } with sexp
+
+  let verify_certificate connection =
+    match Ssl.Connection.peer_certificate connection with
+    | None -> return false
+    | Some (Error _) -> return false
+    | Some (Ok _) -> return true
+
+  let configure ?version ?name ?ca_file ?ca_path ?session ?verify () =
+    { version; name; ca_file; ca_path; session; verify}
+ELSE
+  type config = unit with sexp
+
+  let verify_certificate connection =
+    let _ = connection in return true
+
+  let configure ?version ?name ?ca_file ?ca_path ?session ?verify () =
+    let _, _, _, _, _, _ = (version, name, ca_file, ca_path, session, verify) in
```

Comment:
Wouldn't it be better if `configure` and `verify_certificate` just raised in this clause of the ifdef?